### PR TITLE
1680 Add squeeze SegmentationSaver handler

### PR DESCRIPTION
**Pull request by Nic-Ma**
Fixes #1680 .

### Description
Thanks for @rijobro 's previous enhancement to NiftiSaver about the `squeeze_end_dims` feature, this PR added the support to `SegmentationSaver` handler.

### Status
**Ready**

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [ ] New tests added to cover the changes.
- [ ] Integration tests passed locally by running `./runtests.sh --codeformat --coverage`.
- [ ] Quick tests passed locally by running `./runtests.sh --quick`.
- [ ] In-line docstrings updated.
- [ ] Documentation updated, tested `make html` command in the `docs/` folder.

### CODE REVIEW

File: monai/transforms/io/array.py [hunk 1]
```diff
@@ -206,6 +206,7 @@ class SaveImage(Transform):
             has been moved to the end). So if input is (C,H,W,D), this will be altered to (H,W,D,C), and
             then if C==1, it will be saved as (H,W,D). If D also ==1, it will be saved as (H,W). If false,
             image will always be saved as (H,W,D,C).
+            it's used for NIfTI format only.
```
Comment (rijobro) [inline monai/transforms/io/array.py:209]: good point to add this here. do you think you could add it for the dictionary version, too?

**Review by rijobro — APPROVED** [review]

File: monai/transforms/io/array.py (hunk 1 above)
Comment (Nic-Ma) [reply monai/transforms/io/array.py:209]: Sure, added in the latest commit.
Thanks.

**Pull request merged**